logging/src/main/java/org/jboss/as/logging/stdio/LogContextStdioContextSelector.java

```diff
@@ -50,16 +53,37 @@ public StdioContext getStdioContext() {
         final Logger root = logContext.getLogger(CommonAttributes.ROOT_LOGGER_NAME);
         StdioContext stdioContext = root.getAttachment(STDIO_CONTEXT_ATTACHMENT_KEY);
         if (stdioContext == null) {
-            stdioContext = StdioContext.create(
-                    new NullInputStream(),
-                    new LoggingOutputStream(logContext.getLogger("stdout"), Level.INFO),
-                    new LoggingOutputStream(logContext.getLogger("stderr"), Level.ERROR)
-            );
-            final StdioContext appearing = root.attachIfAbsent(STDIO_CONTEXT_ATTACHMENT_KEY, stdioContext);
+            // Create the StdioContext possibly using a privileged action. This is required for scenarios where a
+            // deployment or library may attempt to write to System.out or System.err.
```

Comment:
@jamezp As we discussed in chat please follow up later with a more detailed comment as it's not obvious why this is needed and why it's safe.

Partly I may be confused because I'm not sure what in the first sentence the "This is required" in the second sentence refers to: "Create the StdioContext" or "using a privileged action".

Reply:
Yes will do. The issue in this case was this blocked me from seeing the other log4j2 security issues because it attempted to write to `System.out` which triggered this error so I lost messages. I should have been more clear in the description.